src/test/java/net/snowflake/client/jdbc/FileUploaderSessionlessTest.java

```diff
@@ -0,0 +1,476 @@
+public class FileUploaderSessionlessTest {
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+  private ObjectMapper mapper = new ObjectMapper();
+
+  private final String exampleS3JsonString =
+      "{\n"
+          + "  \"data\": {\n"
+          + "    \"uploadInfo\": {\n"
+          + "      \"locationType\": \"S3\",\n"
+          + "      \"location\": \"example/location\",\n"
+          + "      \"path\": \"tables/19805757505/\",\n"
+          + "      \"region\": \"us-west-2\",\n"
+          + "      \"storageAccount\": null,\n"
+          + "      \"isClientSideEncrypted\": true,\n"
+          + "      \"creds\": {\n"
+          + "        \"AWS_KEY_ID\": \"EXAMPLE_AWS_KEY_ID\",\n"
+          + "        \"AWS_SECRET_KEY\": \"EXAMPLE_AWS_SECRET_KEY\",\n"
+          + "        \"AWS_TOKEN\": \"EXAMPLE_AWS_TOKEN\",\n"
+          + "        \"AWS_ID\": \"EXAMPLE_AWS_ID\",\n"
+          + "        \"AWS_KEY\": \"EXAMPLE_AWS_KEY\"\n"
+          + "      },\n"
+          + "      \"presignedUrl\": null,\n"
+          + "      \"endPoint\": null\n"
+          + "    },\n"
+          + "    \"src_locations\": [\n"
+          + "      \"/tmp/files/orders_100.csv\"\n"
+          + "    ],\n"
+          + "    \"parallel\": 4,\n"
+          + "    \"threshold\": 209715200,\n"
+          + "    \"autoCompress\": true,\n"
+          + "    \"overwrite\": false,\n"
+          + "    \"sourceCompression\": \"auto_detect\",\n"
+          + "    \"clientShowEncryptionParameter\": true,\n"
+          + "    \"queryId\": \"EXAMPLE_QUERY_ID\",\n"
+          + "    \"encryptionMaterial\": {\n"
+          + "      \"queryStageMasterKey\": \"EXAMPLE_QUERY_STAGE_MASTER_KEY\",\n"
+          + "      \"queryId\": \"EXAMPLE_QUERY_ID\",\n"
+          + "      \"smkId\": 123\n"
+          + "    },\n"
+          + "    \"stageInfo\": {\n"
+          + "      \"locationType\": \"S3\",\n"
+          + "      \"location\": \"stage/location/foo/\",\n"
+          + "      \"path\": \"tables/19805757505/\",\n"
+          + "      \"region\": \"us-west-2\",\n"
+          + "      \"storageAccount\": null,\n"
+          + "      \"isClientSideEncrypted\": true,\n"
+          + "      \"creds\": {\n"
+          + "        \"AWS_KEY_ID\": \"EXAMPLE_AWS_KEY_ID\",\n"
+          + "        \"AWS_SECRET_KEY\": \"EXAMPLE_AWS_SECRET_KEY\",\n"
+          + "        \"AWS_TOKEN\": \"EXAMPLE_AWS_TOKEN\",\n"
+          + "        \"AWS_ID\": \"EXAMPLE_AWS_ID\",\n"
+          + "        \"AWS_KEY\": \"EXAMPLE_AWS_KEY\"\n"
+          + "      },\n"
+          + "      \"presignedUrl\": null,\n"
+          + "      \"endPoint\": null\n"
+          + "    },\n"
+          + "    \"command\": \"UPLOAD\",\n"
+          + "    \"kind\": null,\n"
+          + "    \"operation\": \"Node\"\n"
+          + "  },\n"
+          + "  \"code\": null,\n"
+          + "  \"message\": null,\n"
+          + "  \"success\": true\n"
+          + "}";
```

Comment:
@sfc-gh-tjones I'm not sure whether we have all the info here, do we? I thought we only have the "stageInfo" in the new API?